### ISSUE TITLE
Getting started guide

### ISSUE THREAD

**Issue by codefromthecrypt** [OpenFeign/feign#32]
While there are a couple examples, there's no step-by-step introduction for using feign, or how to make custom decoders, etc.

**Comment by fiunchinho** [comment]
I'm facing problems getting started under an Spring Cloud application. I'm blocked at 

```
com.netflix.hystrix.contrib.javanica.exception.CommandActionExecutionException: feign.codec.DecodeException: Could not read document: Can not deserialize instance of rx.Observable out of START_ARRAY token
```
I assumed that it'd automatically subscribe some decoder to my observable so it could decode the object emitted by the observable. It works fine if I switch the signature from Observable<User> to List<User>, but I'd like to use it asynchronously.
Actually, I'm a bit confused since the documentation from Spring Cloud says to define endpoints like

`@RequestMapping(value = "/users", method = RequestMethod.GET)`

but here the documentation says:

`@RequestLine("GET /users")`

Is there any difference?

**Comment by fiunchinho** [comment]
@adriancole I'll leave an example to illustrate https://gist.github.com/fiunchinho/c250427a2bb0186786c9

Is there a way to subscribe an encoder so I can keep the asynchronous call?

**Comment by spencergibb** [comment]
`@RequestMapping` is from spring cloud Netflix

**Comment by adriancole** [comment]
Yep, this is mostly about features from spring cloud. Can you re-open there?

**Comment by fiunchinho** [comment]
Thank you for your quick replies!
If you see [the gist I posted](https://gist.github.com/fiunchinho/c250427a2bb0186786c9), I'm not using Spring at all. It's the example from Feign documentation, but I get the same error.

**Comment by adriancole** [comment]
Sorry.. I missed the gist paste earlier. Looks legit.

Regardless shouldn't this be a different issue than "getting started
guide". I fear we are cluttering up something unrelated. Mind moving this
to a new issue (and deleting the comments from this issue)?

**Comment by urig** [comment]
Isn't this the getting started guide? https://cloud.spring.io/spring-cloud-netflix/multi/multi_spring-cloud-feign.html (Or at least a good basis for one)

**Comment by spencergibb** [comment]
spring-cloud-netflix is a consumer of openfeign. None of the `@FeignClient` or spring mvc annotation stuff applies here

**Comment by urig** [comment]
Thanks for clarifying.
